infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -171,12 +173,23 @@ public void reset(@Nonnull long epoch) {
         }
     }
 
+    public Tails queryTails(long epoch) {
+        try {
+            CompletableFuture<Tails> cf = new CompletableFuture<>();
+            operationsQueue.add(new BatchWriterOperation(Type.TAILS_QUERY, null,
+                    null, epoch, null, cf));
+            return cf.get();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

Reply:
Will make this change in #1294